src/UnitTests/ValueTransformers.cs

```diff
@@ -0,0 +1,247 @@
+﻿using Shouldly;
+using Xunit;
+
+namespace AutoMapper.UnitTests
+{
+    namespace ValueTransformers
+    {
+        public class BasicTransforming : AutoMapperSpecBase
+        {
+            public class Source
+            {
+                public string Value { get; set; }
+            }
+
+            public class Dest
+            {
+                public string Value { get; set; }
+            }
+
+            protected override MapperConfiguration Configuration { get; } = new MapperConfiguration(cfg =>
+            {
+                cfg.CreateMap<Source, Dest>();
+                cfg.ApplyTransform<string>(dest => dest + " is straight up dope");
+            });
+
+            [Fact]
+            public void Should_transform_value()
+            {
+                var source = new Source
+                {
+                    Value = "Jimmy"
+                };
+                var dest = Mapper.Map<Source, Dest>(source);
+
+                dest.Value.ShouldBe("Jimmy is straight up dope");
+            }
+        }
+
+        public class StackingTransformers : AutoMapperSpecBase
+        {
+            public class Source
+            {
+                public string Value { get; set; }
+            }
+
+            public class Dest
+            {
+                public string Value { get; set; }
+            }
+
+            protected override MapperConfiguration Configuration { get; } = new MapperConfiguration(cfg =>
+            {
+                cfg.CreateMap<Source, Dest>();
+                cfg.ApplyTransform<string>(dest => dest + " is straight up dope");
+                cfg.ApplyTransform<string>(dest => dest + "! No joke!");
```

Comment:
OK I made some changes here lemme know what you think. I couldn't go extension methods all the way because it would have required forcing generic arguments or a weird chaining.

Reply:
The list extension method thing works for profiles, but not for type maps. It breaks the DSL. We could chain through the list, but that seems wrong because we could compose with ForAllMembers or As, for example, right?

Reply:
I think this one is blocking.

Reply:
What's still blocked here? I made the extension method on Profile/Mapper's ValueTransformers, but had to keep AddTransformer in TypeMap. I think that's at least consistent with each's DSL.

Reply:
Yes 😊 I got confused somehow. I thought it was not chaining properly. Sorry about that.